src/tests/ftest/util/telemetry_utils.py

```diff
@@ -1217,13 +1248,21 @@ def _get_data(self, names, info):
                 if name not in data:
                     data[name] = {}
                 for metric in metrics[name]['metrics']:
-                    if 'labels' not in metric or 'value' not in metric:
+                    if 'labels' not in metric or \
+                            ('value' not in metric and 'buckets' not in metric):
                         continue
                     labels = [f'host:{host}']
                     for key, value in metric['labels'].items():
                         labels.append(":".join([str(key), str(value)]))
                     label_key = ",".join(sorted(labels))
-                    data[name][label_key] = metric['value']
+                    if 'value' in metric:
+                        data[name][label_key] = metric['value']
+                    else:
+                        data[name][label_key] = {
+                            'sample_count': metric['sample_count'],
+                            'sample_sum': metric['sample_sum'],
+                            'buckets': metric['buckets'],
+                        }
```

Comment:
This will cause problems with https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/telemetry_utils.py#L1247-L1250 where the `self._data[metric][combined_label]` (i.e. `data[name][label_key]`) value is now a dictionary.

Since the new src/tests/ftest/telemetry/dfs_client_telemetry.py test is not making use of the `self.telemetry.verify_data()` method it wouldn't be an immediate problem, but if future tests verify metrics with sample_count/sample_sum/buckets (or if the metrics for existing tests change to use sample_count/sample_sum/buckets instead of value) will have a problem.

Reply:
OK, so what do you suggest as a solution?

Reply:
@daltonbohning, @phender: Would you prefer that I just remove the ftest changes from this PR and we can come up with a more complete solution in DAOS-16885?

Reply:
IMO, assuming what you have now is functionally correct, let's keep it for basic code path coverage. Then can expand on it in DAOS-16885

Reply:
I have some ideas that could support histogram type metrics as part DAOS-15542.  Since it doesn't seem like it will cause an immediate problem its probably ok.